From 992684d3a91780dd6a38f10dfc157c8187c10cbb Mon Sep 17 00:00:00 2001
From: Josh Wilsdon <josh@wilsdon.ca>
Date: Tue, 22 Jan 2019 09:35:32 -0800
Subject: [PATCH] TRITON-1096 TRITON-804 broke mockcloud Reviewed by: Trent
 Mick <trentm@gmail.com> Approved by: Trent Mick <trentm@gmail.com>

---
 lib/models/server.js | 16 ++++++++++++++--
 package.json         |  2 +-
 2 files changed, 15 insertions(+), 3 deletions(-)

diff --git a/lib/models/server.js b/lib/models/server.js
index 8bdfd36..8ac9892 100644
--- a/lib/models/server.js
+++ b/lib/models/server.js
@@ -1884,7 +1884,13 @@ function (opts) {
             });
         },
         function getCnapiLocationFromSysinfo(wfcb) {
-            serverAdminIp = netconfig.adminIpFromSysinfo(sysinfo);
+            if (sysinfo.hasOwnProperty('CN Agent IP')) {
+                // Allow sysinfo to include an IP that we'll use to connect to
+                // that might be different from the "Admin IP".
+                serverAdminIp = sysinfo['CN Agent IP'];
+            } else {
+                serverAdminIp = netconfig.adminIpFromSysinfo(sysinfo);
+            }
             if (!serverAdminIp) {
                 wfcb(new VError('parsing server ip address in sendTaskReq '
                     + '(No admin NICs detected.)'));
@@ -2087,7 +2093,13 @@ ModelServer.prototype.sendRequest = function (opts, cb) {
             });
         },
         function getCnapiLocationFromSysinfo(wfcb) {
-            serverAdminIp = netconfig.adminIpFromSysinfo(sysinfo);
+            if (sysinfo.hasOwnProperty('CN Agent IP')) {
+                // Allow sysinfo to include an IP that we'll use to connect to
+                // that might be different from the "Admin IP".
+                serverAdminIp = sysinfo['CN Agent IP'];
+            } else {
+                serverAdminIp = netconfig.adminIpFromSysinfo(sysinfo);
+            }
             if (!serverAdminIp) {
                 wfcb(new VError('parsing server ip address in sendTaskReq '
                     + '(No admin NICs detected.)'));
diff --git a/package.json b/package.json
index 073da07..5cc1d16 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cnapi",
   "description": "SmartDataCenter Compute Node API",
-  "version": "1.18.3",
+  "version": "1.18.4",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
-- 
2.21.0

